astropy/coordinates/angles.py

```diff
@@ -640,7 +641,15 @@ def __new__(cls, angle, unit=None, wrap_angle=None, **kwargs):
         self = super().__new__(cls, angle, unit=unit, **kwargs)
         if wrap_angle is None:
             wrap_angle = getattr(angle, 'wrap_angle', self._default_wrap_angle)
-        self.wrap_angle = wrap_angle
+
+        if (issubclass(wrap_angle.__class__, Angle)
+            and wrap_angle.__class__ is not Angle):
+            warnings.warn((
+                    f"`wrap_angle` is <{type(wrap_angle)}>, not <Angle>."
+                    "It is currently converted, but will error in the future."
+                ),
+                DeprecationWarning)
+        self.wrap_angle = Angle(wrap_angle, copy=False)
```

Comment:
Sorry for the annoying comment, but now feel this should really be in a separate PR -- can you pull this out? Fine to leave the docstring change, and the new PR is kind of a bug fix...

Reply:
Done! #11577